### PR TITLE
Change EventList and GTI from Table subclass to attribute

### PR DESCRIPTION
**Pull request by cdeil**
This pull request:
- [x] Changes `EventList` from an `astropy.table.Table` sub-class to a class with an `astropy.table.Table` attribute.
- [x] Add `EventList.select_row_subset` method that dispatches to `EventList.table` and is called everywhere row subset selection occurs.
- [x] Changes `GTI` from an `astropy.table.Table` sub-class to a class with an `astropy.table.Table` attribute.
- [x] Add `EventList` test with a Fermi-LAT event list (make some IACT-specific columns optional, e.g. in print output)
- [x] Some fixes and improvement in `EventListChecker`
- [x] Adapts code using `EventList`
- [x] Misc cleanup and improvements of `EventList` related code

### CODE REVIEW

**Comment by joleroi** [comment]
Please add a test for stacking event lists

**Comment by cdeil** [comment]
@adonath @joleroi I'm not done here adapting the code and test to the new EventList class, and the diff is already messy and hard to review.

But maybe you can have a quick look at this
https://github.com/cdeil/gammapy/blob/ccc016af65defcd3b1143b9826895a1effbbb112/gammapy/data/event_list.py#L73
it's basically this now:
```python
class EventList(object):
    def __init__(self, table):
        self.table = table
```
Any comments?

**Comment by adonath** [comment]
@cdeil I've had a quick look at the changes and I don't have any specific comments. I think introducing the `.table` attribute is a good choice, because it cleans up the `EventList` API and removes inheritance complications with methods such as `.read()`, `.write()` and  `.info()`.

File: gammapy/data/event_list.py [hunk 1]
```diff
@@ -50,20 +54,58 @@ class EventList(Table):
...
+            Filename
+        """
+        filename = make_path(filename)
+        if 'hdu' not in kwargs:
+            kwargs.update(hdu='EVENTS')
+
+        table = Table.read(str(filename), **kwargs)
+        return cls(table=table)
+
+    def info(self, file=None):
```
Comment (joleroi) [inline gammapy/data/event_list.py]: Please change this to ``__str__`` (and add an info method calling it if you like). see http://docs.gammapy.org/en/latest/development/howto.html#object-summary-info-string

Comment (cdeil) [inline gammapy/data/event_list.py]: done

**Comment by cdeil** [comment]
I've finished up this PR and updated the description above.
I'd like to make the 0.5 release today, so some tasks were moved to a future reminder issue: #782

**Comment by cdeil** [comment]
The only fail is this:
https://travis-ci.org/gammapy/gammapy/jobs/176966197#L2449
I'll merge this now, and update the notebook next.

**Pull request merged**